pkg/signup/service/signup_service_test.go

```diff
@@ -245,6 +247,46 @@ func (s *TestSignupServiceSuite) TestGetSignupFailsWithNotFoundThenOtherError()
 	require.Equal(s.T(), "something quite unfortunate happened:something bad", err.Error())
 }
 
+func (s *TestSignupServiceSuite) TestSignupNoSpaces() {
+	s.ServiceConfiguration(TestNamespace, true, "", 5)
+
+	// given
+	userID, err := uuid.NewV4()
+	require.NoError(s.T(), err)
+
+	rr := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(rr)
+	ctx.Set(context.UsernameKey, "jsmith")
+	ctx.Set(context.SubKey, userID.String())
+	ctx.Set(context.OriginalSubKey, "original-sub-value")
+	ctx.Set(context.EmailKey, "jsmith@gmail.com")
+	ctx.Set(context.GivenNameKey, "jane")
+	ctx.Set(context.FamilyNameKey, "doe")
+	ctx.Set(context.CompanyKey, "red hat")
+	ctx.Request, _ = http.NewRequest("POST", "/?no-space=true", bytes.NewBufferString(""))
+
+	// when
+	userSignup, err := s.Application.SignupService().Signup(ctx)
+
+	// then
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), userSignup)
+
+	gvk, err := apiutil.GVKForObject(userSignup, s.FakeUserSignupClient.Scheme)
+	require.NoError(s.T(), err)
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+
+	values, err := s.FakeUserSignupClient.Tracker.List(gvr, gvk, configuration.Namespace())
+	require.NoError(s.T(), err)
+
+	userSignups := values.(*toolchainv1alpha1.UserSignupList)
+	require.NotEmpty(s.T(), userSignups.Items)
+	require.Len(s.T(), userSignups.Items, 1)
+
+	val := userSignups.Items[0]
+	require.Equal(s.T(), "true", val.Annotations[toolchainv1alpha1.SkipAutoCreateSpaceAnnotationKey]) // skip auto create space annotation is set
```

Comment:
Let's also add a check to another test which doesn't have `?no-space=true` in the request that the annotation is not present/set to true.

Reply:
Done in 66700dadb45ea1d5c8bdc28f922853b0d94f0630